File: Tutorial4-4ComponentBuildersMultipleModules/src/main/java/com/example/tutorial4_4componentbuildersmultiplemodules/di/AppModuleWithBind.java
package com.example.tutorial4_4componentbuildersmultiplemodules.di;

import android.app.Application;
import android.content.Context;
import android.content.SharedPreferences;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class AppModuleWithBind {

    @Provides
    @Singleton
    public SharedPreferences providePreferences(Application application) {
        return application.getSharedPreferences("data",
                Context.MODE_PRIVATE);
    }
}

File: Tutorial4-4ComponentBuildersMultipleModules/src/main/java/com/example/tutorial4_4componentbuildersmultiplemodules/di/AppModuleWithBind.kt
package com.example.tutorial4_4componentbuildersmultiplemodules.di

import android.app.Application
import android.content.Context
import android.content.SharedPreferences
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class AppModuleWithBind {

    @Provides
    @Singleton
    fun providePreferences(application: Application): SharedPreferences {
        return application.getSharedPreferences("data",
                Context.MODE_PRIVATE)
    }

}